nncf/experimental/torch/fx/model_transformer.py

```diff
@@ -0,0 +1,141 @@
+
+    def __init__(self, model: torch.fx.GraphModule):
+        super().__init__(model)
+
+        self._command_transformation_ordered_pairs = [
+            (FXApplyTransformationCommand, self._apply_transformation),
+            (PTModelExtractionCommand, self._apply_model_extraction),
+        ]
+
+    def transform(self, transformation_layout: PTTransformationLayout) -> torch.fx.GraphModule:
+        """
+        Transforms the target model according to given transformation layout.
+
+        :param transformation_layout: Given transformation layout.
+        :return: Target model transformered according to the given transformation layout.
+        """
+        # TODO(dlyakhov): Manage priorities of transformations.
+        transformations = transformation_layout.transformations
+        aggregated_transformations = defaultdict(list)
+        for transformation in transformations:
+            aggregated_transformations[transformation.__class__].append(transformation)
+
+        model = self._model
+        for transformation_cls, transformation_fn in self._command_transformation_ordered_pairs:
+            transformations = aggregated_transformations[transformation_cls]
+            if transformations:
+                model = transformation_fn(model, transformations)
+
+        # Do not use model.graph.eliminate_dead_code()
+        # because the computational statistics code
+        # is interpolated as dead code.
+        model.recompile()
+        return model
+
+    @staticmethod
+    def _apply_model_extraction(
+        model: torch.fx.GraphModule,
+        transformations: List[PTModelExtractionCommand],
+    ) -> torch.fx.GraphModule:
+        """
+        Returns a submodel extracted from the given model by the given transformation.
+
+        :param model: Given model.
+        :param transformations: List of one transformation which specifies
+            how to retrieve a submodule from the model. In case list contains
+            more than one element this function raises an assert.
+        :return: Returns a submodel extracted from the given model by the given transformation.
+        """
+        transformation = transformations[-1]
+        assert len(transformation.input_node_names) == 1
+        assert transformation.input_node_names == transformation.output_node_names
+        node_name = transformation.input_node_names[0]
+
+        tags = ["before", "extracted", "after"]
+        i = 0
+        for node in model.graph.nodes:
+            if node.name == node_name:
+                node.tag = tags[1]
+                weights = [node.all_input_nodes[1]]
+                while weights:
+                    w_node = weights.pop()
+                    assert w_node.tag in tags[0:2]
+                    w_node.tag = tags[1]
+                    weights.extend(w_node.all_input_nodes)
+                i = 2
+                continue
+            node.tag = tags[i]
+
+        splitted_gm = split_by_tags(model, tags)
```

Comment:
Looks like some additional code still required to utilize the `lift_subgraph_as_module` function. I've added a todo string, we should reimplement the `split_by_tags` function in case of memory consumption limitations